en/sensor/rangefinders.md

```diff
@@ -0,0 +1,180 @@
+
+The sensors are enabled using the parameter [SENS_EN_TRANGER](../advanced_config/parameter_reference.md#SENS_EN_TRANGER) (you can set the type of sensor or that PX4 should auto-detect the type).
+
+> **Info** The *Terranger One* is used in the [Qualcomm Snapdragon Flight](../flight_controller/snapdragon_flight.md). 
+
+### uLanding Radar
+
+The *Aerotenna* [uLanding](https://aerotenna.com/sensors/#ulanding) altimeter is compact microwave rangefinder that has been optimised for use on UAVs. It has a sensing range of 45m. A particular advantages of this product are that it can operate effectively in all weather conditions and over all terrain types (including water).
+
+The uLanding Radar is not present in "most" firmware by default and must be started by updating a configuration file (rather than by a parameter). More information can be found here: [uLanding Radar](../sensor/ulanding_radar.md).
+
+
+### Other
+
+PX4 also supports the Bebop rangefinder.
+
+
+## Rangefinder Configuration
+
+### Setup
+
+The rangefinder is configured using [EKF2\_RNG\_*](../advanced_config/parameter_reference.md#EKF2_RNG_POS_X) parameters. These configure information about the offset of the rangefinder from the centre of the vehicle body, the approximate delay of data reaching the estimator from the sensor, etc.
+
+### Enable/Use
+
+The rangefinder can be enabled in two ways:
+1. Set [EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) to *Range finder* (`2`). This makes the rangefinder the primary source of height estimation (the default altitude sensor is the barometer).
+1. Set [EKF2_RNG_AID](../advanced_config/parameter_reference.md#EKF2_RNG_AID) to `1`. This makes the vehicle use the rangefinder as the primary source when it is safe to use, but will otherwise use the sensor specified in `EKF2_HGT_MODE`.
+   * Specifically, the rangefinder is enabled when: 
+     * velocity < [EKF2_RNG_A_VMAX](../advanced_config/parameter_reference.md#EKF2_RNG_A_VMAX)
+     * distance to ground < [EKF2_RNG_A_HMAX](../advanced_config/parameter_reference.md#EKF2_RNG_A_HMAX)
+   * Other parameters affecting "Range aid" are prefixed with `EKF2\_RNG\_A\_`.
+
+## Testing
+
+The easiest way to test the rangefinder is to vary the range and compare to the values detected by PX4. The sections below show some approaches to getting the measured range.
+
+### QGC MAVLink Inspector
+
+The *QGroundControl MAVLink Inspector* tool lets you check the altitude sent from the vehicle. If you set the vehicle to use a rangefinder as the main height source, then this should reflect the sensor values:
+
+1. Make the rangefinder the main height source (set parameter `EKF2_HGT_MODE` to Range finder (`2`))
+1. Open the menu **Widgets > MAVLink Inspector**:
+
+   ![Menu for QGC MAVLink Inspector](../../assets/qgc/menu_mavlink_inspector.png)
+1. Select the target vehicle and then the `ALTITUDE` message. Observe the value as shown:

```

Comment:
In the inspector you should also see a `DISTANCE_SENSOR` message (without having to change the height source).
Or even better with `Widgets -> Analyze`, you can see a graph in real-time:
![screenshot_20171130_100139](https://user-images.githubusercontent.com/281593/33422330-a1ce0bf6-d5b5-11e7-9a54-f160d76a5e08.png)

Reply:
Nope, that message isn't visible in either view. I get the small subset below. Perhaps the set is different in a daily build?
Actually more likely I just don't have a distance sensor on this simulated vehicle ?

![image](https://user-images.githubusercontent.com/5368500/33423020-be9e9b28-d60b-11e7-9000-3cea79f1dae3.png)

Reply:
That's the Analyse tool Beat is talking about and not the Mavlink Inspector

Reply:
The subset of options in Analyse is even smaller.

Reply:
You're running `make posix_sitl_default gazebo_iris_opt_flow`?

Reply:
@hamishwillee the set of sent/displayed messages is defined by the Firmware (the configured mavlink streams), and only visible if there's a distance sensor attached (i.e. not with jMAVSim).